### ISSUE TITLE
native gRPC failed invocations rate metric

### ISSUE THREAD

**Issue by murlock1000** [grafana/k6#3164]
### Feature Description

Implement native metric for the rate of failed gRPC invocations, similar to http_req_failed.

### Suggested Solution (optional)

Rate calculated as follows:

```
response = client.invoke()
grpc_req_failed.add(response.status === grpc.StatusOK);
```

**Comment by olegbespalov** [comment]
Hi @murlock1000 !

Could you please describe the use case? I'm trying to get a better understanding of the rationale why you want to see the native metric instead of using the custom one. Thanks!

**Comment by murlock1000** [comment]
I would expect feature parity between http and grpc metrics. It would also be used for setting up native instead of custom thresholds.

**Comment by brendenehlers** [comment]
hey folks, any update on this? we're missing the failed request metrics in our reports and it's posing a problem. happy to provide additional context if needed.

**Comment by codebien** [comment]
>happy to provide additional context if needed

@brendenehlers Yes, we need it to have a clear idea of the problem and ideally prioritize highly.

**Comment by brendenehlers** [comment]
when load testing a traditional http endpoint, k6 provides both the `http_req_failed` and `http_req_duration` metrics out of the box. it looks like currently only the `grpc_req_duration` metric is provided when load testing a grpc endpoint. what i'm asking for is feature parity between http load testing and grpc load testing, where both `grpc_req_failed` and `grpc_req_duration` are provided metrics.

links:
[default http metrics](https://grafana.com/docs/k6/latest/using-k6/thresholds/#thresholds-example-for-http-errors-and-response-duration)

[closest thing to a reference doc for provided grpc metrics](https://grafana.com/docs/k6/latest/testing-guides/performance-testing-grpc-services/#run-the-test)

**Comment by codebien** [comment]
Hey @brendenehlers,
sorry for the late reply. How do you expect to count failed requests? My assumption is that you want to count a request as failed when it is one of these following [error status codes](https://grpc.io/docs/guides/error/#error-status-codes)?

Is it sufficient or did you have a different idea for them?

**Comment by brendenehlers** [comment]
@codebien yeah exactly everything gRPC classifies as an "error" is what I would expect to count as failed.
